### DISCUSSION TITLE
Should the verbose parameter in BaseGridSearch be a bool instead of an int?

### DISCUSSION THREAD

**Discussion by edzWH** [alan-turing-institute/sktime#7326] [General]
When using ForecastingOptunaSearchCV I wanted to increase the verbosity to see how many iterations of CV params had been back-tested, so I could judge how long left the CV would take. 

When reading the docs it states "Controls the verbosity: the higher, the more messages.", which isn't particularly clear.

Looking at references to verbose in the _tune.py file the only reference I can find is the following which is used in the BaseGridSearch and other grid search classes. Which prints the number of splits and candidate params. 

'if self.verbose > 0:
                n_candidates = len(candidate_params)
                n_splits = cv.get_n_splits(y)
                print(
                    f"Fitting {n_splits} folds for each of {n_candidates} candidates,"
                    f" totalling {n_candidates * n_splits} fits"
                )'

If there is only 1 check on verbose being greater than 0 should this be set to a boolean value? Or am I missing something?

`

**Comment by fkiraly** [comment]
Hm, I think this is the level of verbosity, similar to how stacklevel in `warn` is managed.

Though I would not know off the top of my head if there are any changes at a level bigger than 0.

**Reply by MarkusSagen** [reply]
@edzWH 
This is likely to have feature parity with other libraries, s.a. lightgbm and sklearn
This is just one example in sklearn: https://github.com/scikit-learn/scikit-learn/blob/b7d45ad568869df6676b72ea53b57e971ef11d5f/sklearn/mixture/_bayesian_mixture.py#L192

**Reply by edzWH** [reply]
This is how I understand it, although from what I can see it doesn't mirror the same functionality.

It could follow something similar to:

0 = Nothing printed
1 = Current functionality
2 = More verbose information for each iteration, e.g. Iteration number, params tested and score 

I've never contributed before but I could have a look into this if it's something which is useful.
